cumulusci/tasks/bulkdata/generate_from_yaml.py

```diff
@@ -55,19 +87,79 @@ def _generate_data(self, db_url, mapping_file_path, num_records, current_batch_n
         else:
             self.mappings = {}
         session, engine, base = self.init_db(db_url, self.mappings)
+        self.logger.info(f"Generating batch {current_batch_num} with {num_records}")
         self.generate_data(session, engine, base, num_records, current_batch_num)
         session.commit()
         session.close()
 
+    def default_continuation_file_path(self):
+        return Path(self.working_directory) / "continuation.yml"
+
+    def get_old_continuation_file(self) -> Optional[Path]:
+        """Use a continuation file if specified or look for one in the working directory
+
+        Return None if no file can be found.
+        """
+        old_continuation_file = self.options.get("continuation_file")
+
+        if old_continuation_file:
+            old_continuation_file = Path(old_continuation_file)
+            if not old_continuation_file.exists():
+                raise TaskOptionsError(f"{old_continuation_file} does not exist")
+        elif self.working_directory:
+            path = self.default_continuation_file_path()
+            if path.exists():
+                old_continuation_file = path
+
+        return old_continuation_file
+
+    def open_new_continuation_file(self) -> Optional[TextIO]:
+        """Create a continuation file based on config or working directory
+
+        Return None if there is no config nor working directory.
+        """
+        if self.options.get("generate_continuation_file"):
+            new_continuation_file = open(
+                self.options["generate_continuation_file"], "w+"
+            )
+        elif self.working_directory:
+            new_continuation_file = open(
+                Path(self.working_directory) / "continuation_next.yml", "w+"
+            )
+        else:
+            new_continuation_file = None
+        return new_continuation_file
+
     def generate_data(self, session, engine, base, num_records, current_batch_num):
         output_stream = SqlOutputStream.from_connection(
             session, engine, base, self.mappings
         )
+        old_continuation_file = self.get_old_continuation_file()
+        if old_continuation_file:
+            # reopen to ensure file pointer is at starting point
+            old_continuation_file = open(old_continuation_file, "r")
+        new_continuation_file = self.open_new_continuation_file()
+
         with open(self.yaml_file) as open_yaml_file:
             summary = generate(
-                open_yaml_file, self.num_records, self.vars, output_stream
+                open_yaml_file=open_yaml_file,
+                user_options=self.vars,
+                output_stream=output_stream,
+                stopping_criteria=self.stopping_criteria,
+                continuation_file=old_continuation_file,
+                generate_continuation_file=new_continuation_file,
             )
             output_stream.close()
+
+            if (
```

Comment:
I don't think I follow the full lifecycle of the continuation file (where each is created and how it's moved through the load process). Let's talk that through today.

Reply:
Maybe this will help:

7fd12dac